base/common/src/org/dogtagpki/nss/NSSDatabase.java

```diff
@@ -343,13 +371,384 @@ public void addPEMCertificate(
         }
     }
 
+    public void addBasicConstraintsExtension(
+            List<String> cmd,
+            PrintWriter out,
```

Comment:
Why is this parameter named `out`? I'd name it `stdin`.

Reply:
(It is confusing, because it is an output of the `void` function, but you're using it to provide _input_ to the `cmd` when it is finally executed)>

Reply:
Technically this code is writing an output string, hence the `out` param name, but later the string will be used as a standard input for `certutil`. I don't mind changing it if you prefer `stdin`. Note that this code will be removed as soon as we switch to JSS.